index.js

```diff
@@ -54,25 +143,67 @@ MediaSourceStream.prototype._write = function (chunk, encoding, cb) {
     return cb(new Error('Cannot append buffer while source buffer updating'))
   }
 
-  self._sourceBuffer.appendBuffer(chunk)
+  try {
+    self._sourceBuffer.appendBuffer(toArrayBuffer(chunk))
+  } catch (err) {
+    self.emit('error', err)
```

Comment:
This seems is a critical error, right? If one chunk of data fails to be written, later chunks won't make any sense. I think we should change this to `self.destroy(err)` so the whole stream is cleaned up and `'close'` is emitted (this calls `unpipe` which stops further writes).